integration_test.go

```diff
@@ -0,0 +1,189 @@
+// +build all integration
+
+package gocql
+
+// This file groups integration tests where Cassandra has to be set up with some special integration variables
+import (
+	"reflect"
+	"testing"
+	"time"
+)
+
+// TestAuthentication verifies that gocql will work with a host configured to only accept authenticated connections
+func TestAuthentication(t *testing.T) {
+
+	if *flagProto < 2 {
+		t.Skip("Authentication is not supported with protocol < 2")
+	}
+
+	if !*flagRunAuthTest {
+		t.Skip("Authentication is not configured in the target cluster")
+	}
+
+	cluster := createCluster()
+
+	cluster.Authenticator = PasswordAuthenticator{
+		Username: "cassandra",
+		Password: "cassandra",
+	}
+
+	session, err := cluster.CreateSession()
+
+	if err != nil {
+		t.Fatalf("Authentication error: %s", err)
+	}
+
+	session.Close()
+}
+
+func TestGetHosts(t *testing.T) {
+	cluster := createCluster()
+	session := createSessionFromCluster(cluster, t)
+
+	hosts, partitioner, err := session.hostSource.GetHosts()
+
+	assertTrue(t, "err == nil", err == nil)
+	assertEqual(t, "len(hosts)", len(clusterHosts), len(hosts))
+	assertTrue(t, "len(partitioner) != 0", len(partitioner) != 0)
+}
+
+//TestRingDiscovery makes sure that you can autodiscover other cluster members when you seed a cluster config with just one node
+func TestRingDiscovery(t *testing.T) {
+	cluster := createCluster()
+	cluster.Hosts = clusterHosts[:1]
+
+	session := createSessionFromCluster(cluster, t)
+	defer session.Close()
+
+	if *clusterSize > 1 {
+		// wait for autodiscovery to update the pool with the list of known hosts
+		time.Sleep(*flagAutoWait)
+	}
+
+	session.pool.mu.RLock()
+	defer session.pool.mu.RUnlock()
+	size := len(session.pool.hostConnPools)
+
+	if *clusterSize != size {
+		for p, pool := range session.pool.hostConnPools {
+			t.Logf("p=%q host=%v ips=%s", p, pool.host, pool.host.ConnectAddress().String())
+
+		}
+		t.Errorf("Expected a cluster size of %d, but actual size was %d", *clusterSize, size)
+	}
+}
+
+func TestWriteFailure(t *testing.T) {
```

Comment:
So, for the versions where we do not set the appropriate JVM extra option this fails. So we either need to check a flag here (i.e., maybe check that the JVM option is set?) or set the JVM flag for all versions.